### PR TITLE
Python Script Plugin validator

### PR DESCRIPTION
**Pull request by lmilby-r7**
## Description

Detects usage of the Python 3 Script plugin and warns of its use, giving the step name its used in.

## Testing

```
[*] Running Integration Validators...
[*] Executing validator WorkflowPythonScriptUseValidator
[*] Workflow failed validation! The following validation errors occurred:

Validator "WorkflowPythonScriptUseValidator" failed! 
	Cause: Python Plugin use found in these steps
		[*] Step Name - Python script used to take over the world
		[*] Step Name - Second attempt at trying to take over the world with a python script
	Please review these steps carefully before use
```

### CODE REVIEW

File: icon_validator/workflow/model.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+from typing import List, Optional, Type, Union
+from dataclasses import dataclass, field
+
+
+class Templates:
+    """
+    Templates is a model thats used for managing the template development lifecycle
+    """
+
+    # TODO Create template struct
```
Comment (pmara-r7) [inline icon_validator/workflow/model.py:10]: Why is this included here? Is it used? (I'm also not familiar with what a Template is so might be obvious)

File: icon_validator/workflow/model.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        for step in plugin_steps:
+            plugins.append({
+                "plugin_name": step["plugin"]["name"],
+                "node_name": step["name"]
+            })
+        return plugins
+
+
+@dataclass
+class Kom:
```
Comment (pmara-r7) [inline icon_validator/workflow/model.py:116]: What is "Kom"?

**Review by pmara-r7 — APPROVED** [review]
Some slightly confusing names/naming conventions (between calling some things "Kom" vs. Komand, but I'm comfortable approving either way

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: icon_validator/workflow/model.py (hunk 1 above)
Comment (lmilby-r7) [reply icon_validator/workflow/model.py:10]: Oh this was existing code that I pulled over. Ill. get this cleaned up

File: icon_validator/workflow/model.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    name: str = ""
+    title: str = ""
+    description: str = ""
+    version: str = ""
+    vendor: str = ""
+    support: str = ""
+
+
+@dataclass
+class Plugin:
```
[the remark this answers is not in the document]
Comment (lmilby-r7) [reply icon_validator/workflow/model.py:150]: I can but before I put more work in on this model I would like to see if can leverage pydantic and recreate this model

File: icon_validator/rules/workflow_validators/workflow_python_script_use_validator.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        wf = dict()
+        d = spec.directory
+        for file_name in os.listdir(d):
+            if file_name.endswith(".icon"):
+                try:
+                    wf = read_workflow(spec=spec, file_name=file_name)
+                except json.JSONDecodeError:
+                    raise ValidationException(
+                        "ICON file is not in JSON format try exporting the workflow file again"
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_python_script_use_validator.py:43]:
```suggestion
                        "ICON file is not in JSON format, try exporting the workflow file again"
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by aholland-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]
